src/Dapr.Actors/Runtime/ActorManager.cs

```diff
@@ -340,6 +339,13 @@ private async Task<T> DispatchInternalAsync<T>(ActorId actorId, ActorMethodConte
             T retval;
             try
             {
+                // Set the state context of the request, if possible.
+                var stateManager = state.Actor.StateManager;
+                if (typeof(IActorContextualState).IsAssignableFrom(stateManager.GetType()))
```

Comment:
You can do this a lot simpler. 

- `Type.IsAssignableFrom` is only needed when you have a two types to compare, you have a type and a value.
- Pattern matching (relatively new in C#) makes this really nice

```C#
if (state.Actor.StateManager is IActorContextualState contextualStateManager)
{
    contextualStateManager.SetStateContext(...)
}
```

Reply:
Such fancy language features!